src/Orleans.Runtime/Silo/Silo.cs

```diff
@@ -811,15 +811,22 @@ private void HandleProcessExit(object sender, EventArgs e)
             // NOTE: We need to minimize the amount of processing occurring on this code path -- we only have under approx 2-3 seconds before process exit will occur
             logger.Warn(ErrorCode.Runtime_Error_100220, "Process is exiting");
             
+            var cancellationSource = new CancellationTokenSource();
             lock (lockable)
             {
                 if (!this.SystemStatus.Equals(SystemStatus.Running)) return;
-                    
+
                 this.SystemStatus = SystemStatus.Stopping;
+
+                // force a non-graceful stop
+                cancellationSource.Cancel();
+                this.siloLifecycle.OnStop(cancellationSource.Token);  // don't wait for it to stop
```

Comment:
Why this `OnStop` call here when `OnStop` will be called via `StopAsync` below?

Reply:
Because the OnStop after call will get into the mode that just waits for State to Terminate, but nothing signals the lifecycle to stop.